frontend/talentsearch/src/js/components/governmentInfoForm/GovernmentInfoForm.tsx

```diff
@@ -0,0 +1,431 @@
+  const handleUpdateUser = (id: string, data: UpdateUserAsUserInput) =>
+    executeMutation({
+      id,
+      user: data,
+    }).then((result) => {
+      if (result.data?.updateUserAsUser) {
+        return result.data.updateUserAsUser;
+      }
+      return Promise.reject(result.error);
+    });
+  const formValuesToSubmitData = (
+    values: FormValues,
+  ): UpdateUserAsUserInput => ({
+    isGovEmployee: values.govEmployeeYesNo === "yes",
+    interestedInLaterOrSecondment: values.lateralDeployBool,
+    currentClassification: {
+      connect: classificationFormToId(
+        values.currentClassificationGroup,
+        values.currentClassificationLevel,
+      ),
+    },
+  });
+  const id = holder ? holder.id : "";
+  const onSubmit: SubmitHandler<FormValues> = async (data: FormValues) => {
+    await handleUpdateUser(id, formValuesToSubmitData(data))
+      .then(() => {
+        navigate(paths.profile());
+        toast.success(
+          intl.formatMessage({
+            defaultMessage: "User updated successfully!",
+            description:
+              "Message displayed to user after user is updated successfully.",
+          }),
+        );
+      })
+      .catch(() => {
+        toast.error(
+          intl.formatMessage({
+            defaultMessage: "Error: updating user failed",
+            description:
+              "Message displayed to user after user fails to get updated.",
+          }),
+        );
+      });
+  };
+
+  if (fetchingLookupData || fetchingMe) {
+    return (
+      <p>
+        {intl.formatMessage({
+          defaultMessage: "Loading...",
+          description: "Loading message",
+        })}
+      </p>
+    );
+  }
+
+  return (
+    <div>
+      <ProfileFormWrapper
+        description={intl.formatMessage({
+          defaultMessage:
+            "Please indicate if you are currently an employee in the Government of Canada.",
+          description:
+            "Description blurb for Profile Form Wrapper in the Government Information Form",
+        })}
+        title={intl.formatMessage({
+          defaultMessage: "Government Information",
+          description:
+            "Title for Profile Form Wrapper in Government Information Form",
+        })}
+        crumbs={[
+          {
+            title: intl.formatMessage({
+              defaultMessage: "Government Information",
+              description:
+                "Display Text for Government Information Form Page Link",
+            }),
+          },
+        ]}
+      >
+        <FormProvider {...methods}>
+          <form onSubmit={handleSubmit(onSubmit)}>
+            <GovernmentInfoForm
+              classifications={classifications}
+              storedValues={previousData}
+            />
+            <ProfileFormFooter
+              mode="saveButton"
+              handleSave={handleSubmit(onSubmit)}
+            />
+          </form>
+        </FormProvider>
```

Comment:
Its a good ponit to me. I was thinking FormProvider should part of the form component. But I don't understand why we need  the handleSubmit in both form and the save button. I am thinking  since the GovernmentInfoForm component is inside the FormProvider component, ProfileFormWrapper component should end before the FormProvider component end. Am I wrong? or either way is correct?

Reply:
That's a good point. We may not need to pass a handler to the ProfileFormFooter button at all!

Reply:
Good call! The SaveButton, and ProfileFormWrapper, should actually not accept a submit handler at all. I've made that refactor in this PR.